yt/frontends/ramses/data_structures.py

```diff
@@ -591,28 +678,33 @@ def __init__(
             max_level, max_level_convention
         )
 
-        # Infer if the output is organized in groups
-        root_folder, group_folder = os.path.split(os.path.split(filename)[0])
+        file_handler = RAMSESFileSanitizer(filename)
+
+        # This should not happen, but let's check nonetheless.
+        if not file_handler.is_valid:
+            raise RuntimeError(
```

Comment:
RuntimeError should be reserved to when no other builtin error is relevant
```suggestion
            raise ValueError(
```